doc(database::informix) (#3962)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-databases-informix.md

| Name   | Unit |
|:-------|:-----|
| ovlock | N/A  |

</TabItem>
<TabItem value="Logfiles-Usage" label="Logfiles-Usage">

| Name             | Unit |
|:-----------------|:-----|
| used.#logfile-id | %    |

</TabItem>
<TabItem value="Longtxs" label="Longtxs">

| Name  | Unit  |
|:------|:------|
| count | count |

</TabItem>
<TabItem value="Sessions" label="Sessions">

| Name     | Unit |
|:---------|:-----|
| sessions | N/A  |

</TabItem>
<TabItem value="Sql-Statement" label="Sql-Statement">

| Name                              | Unit |
|:----------------------------------|:-----|
| value                             | N/A  |
| sqlrequest.execution.time.seconds | s    |

</TabItem>
<TabItem value="Table-Locks-*" label="Table-Locks-*">

| Name                         | Unit  |
|:-----------------------------|:------|
| db\_deadlks\dbname           | count |
| db\_lockwts\dbname           | count |
| db\_lockreqs\dbname          | count |
| db\_lktouts\dbname           | count |
| tbl\_deadlks\dbname.tabname  | count |
| tbl\_lockwts\dbname.tabname  | count |
| tbl\_lockreqs\dbname.tabname | count |
| tbl\_lktouts\dbname.tabname  | count |

</TabItem>
</Tabs>

## Prerequisites

> You need to download "Informix Client Software Development Kit" on [IBM website](https://www.ibm.com/support/pages/informix-client-software-development-kit-client-sdk-and-informix-connect-system-requirements).

Reviewer: ```suggestion
You need to download the "Informix Client Software Development Kit" on the [IBM website](https://www.ibm.com/support/pages/informix-client-software-development-kit-client-sdk-and-informix-connect-system-requirements).
```

> Install with the following procedure:

Reviewer: ```suggestion
Install the SDK with the following procedure:
```

> 1. Copy archive on the poller

Reviewer: ```suggestion
1. Copy the archive on the poller.
```

> 2. Create an user informix

Reviewer: ```suggestion
2. Create an **informix** user:
```

```
useradd informix chmod 775 /home/informix
``` 

> 3. Need to install Sun JRE (as explain in the IBM process).

Reviewer: ```suggestion
3. Install Sun JRE (as explained in the IBM procedure).
```

> 4. Install SDK (choose Typical Installation): 

Reviewer: ```suggestion
4. Install the SDK (select**Typical Installation**): 
```

```
./installclientsdk -javahome /usr/java/jreXXXX/ ....
```

> Specify a directory or press Enter to accept the default directory. 

Reviewer: ```suggestion

5. Specify a directory or press **Enter** to accept the default directory. 
```

Directory Name: `[/root/informix/sdkclient/] /home/informix/sdkclient` 

### Perl DBD Informix

> To compile DBD Informix, you need an access to an Informix Database, run:

Reviewer: ```suggestion

1. To compile Informix DBD, you need access to an Informix Database. Run:

```

| EXTRAOPTIONS     | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --only-databases --verbose |             |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_informix.pl \
	--plugin=database::informix::sql::plugin \
	--host=10.0.0.1 \
	--username='' \

> 	--password='' \

Reviewer: username and password

Author: Not mandatory

Install the SDK with the following procedure:
1. Copy the archive on the poller.
2. Create an **informix** user:

```
useradd informix chmod 775 /home/informix
``` 

3. Install Sun JRE (as explained in the IBM procedure).
4. Install the SDK (select**Typical Installation**): 

```
./installclientsdk -javahome /usr/java/jreXXXX/ ....
```

5. Specify a directory or press **Enter** to accept the default directory. 
Directory Name: `[/root/informix/sdkclient/] /home/informix/sdkclient` 

### Perl DBD Informix

1. To compile Informix DBD, you need access to an Informix Database. Run:

```
cd /usr/local/src
wget http://search.cpan.org/CPAN/authors/id/J/JO/JOHNL/DBD-Informix-2013.0521.tar.gz
tar xzf DBD-Informix-2013.0521.tar.gz
cd DBD-Informix-2013.0521
export INFORMIXDIR=/home/informix/sdkclient
export LD_LIBRARY_PATH=$ORACLE_HOME/lib
export PATH=${PATH}:/home/informix/sdkclient/bin
export LD_LIBRARY_PATH=/home/informix/sdkclient/lib/esql/:/home/informix/sdkclient/lib/
export DBD_INFORMIX_USERNAME=root
export DBD_INFORMIX_PASSWORD=xxxx # export DBD_INFORMIX_DATABASE=xxxx
```

> Set Informix Instance in `/home/informix/sdkclient/etc/sqlhosts` file:

Reviewer: ```suggestion
Set a value for **Informix Instance** in the `/home/informix/sdkclient/etc/sqlhosts` file:
```
